### PR TITLE
Ensure we capture baggage when capturing Errors during unsampled transactions

### PR DESCRIPTION
**Pull request by Mpdreamz**

### CODE REVIEW

File: src/Elastic.Apm/Model/Error.cs [hunk 1]
```diff
@@ -55,19 +55,23 @@ private Error(Transaction transaction, string parentId, IApmLogger loggerArg, Di
...
 
-		private void CheckAndCaptureBaggage()
+		private void CheckAndCaptureBaggage(Transaction transaction)
 		{
 			if (Activity.Current == null || !Activity.Current.Baggage.Any())
 				return;
 
+			//if context was not set prior we set it now to ensure we capture baggage for errors
+			//occuring during unsampled transactions
+			Context ??= transaction.Context.DeepCopy();
```
Comment (gregkalapos) [inline src/Elastic.Apm/Model/Error.cs:73]: Did the line below with `var labels = Context.InternalLabels.Value;` fail with a `NullReferenceException`? Or what was the behaviour before?

Comment (Mpdreamz) [reply src/Elastic.Apm/Model/Error.cs:73]: Context is null because we never copy it from transaction when its an unsampled transaction.

**Review by gregkalapos — APPROVED** [review]

File: src/Elastic.Apm/Model/Error.cs (hunk 1 above)
Comment (Mpdreamz) [reply src/Elastic.Apm/Model/Error.cs:73]: Thought about it but this way we only do it if there is actual baggage to set :)

Comment (gregkalapos) [reply src/Elastic.Apm/Model/Error.cs:73]: > Thought about it but this way we only do it if there is actual baggage to set :)

Yeah, that's a good point.

Comment (aheubusch) [inline src/Elastic.Apm/Model/Error.cs:73]: Requires a null check. `transaction` can be null.

**Pull request merged**